wiring/src/spark_wiring_ble.cpp

```diff
@@ -0,0 +1,1756 @@
+        }
+        if (ret == SYSTEM_ERROR_NONE) {
+            ret = advertise();
+        }
+        return ret;
+    }
+
+    int stopAdvertising(void) const {
+        return ble_gap_stop_advertising();
+    }
+
+    void broadcasterProcessStopped(void) {
+        return;
+    }
+};
+
+
+/*******************************************************
+ * BleObserverImpl definition
+ */
+class BleObserverImpl {
+public:
+    size_t targetCount;
+    OnScanResultCallback callback;
+    BleScanResult* results;
+    size_t count;
+
+    BleObserverImpl() : targetCount(0), callback(nullptr), results(nullptr), count(0) {}
+    ~BleObserverImpl() {}
+
+    int scan(OnScanResultCallback callback) {
+        this->callback = callback;
+        count = 0;
+        ble_gap_start_scan(observerProcessScanResult, this, nullptr);
+        return count;
+    }
+
+    int scan(OnScanResultCallback callback, uint16_t timeout) {
+        hal_ble_scan_params_t params;
+        ble_gap_get_scan_parameters(&params, nullptr);
+        params.timeout = timeout;
+        ble_gap_set_scan_parameters(&params, nullptr);
+        return scan(callback);
+    }
+
+    int scan(BleScanResult* results, size_t resultCount) {
+        this->results = results;
+        targetCount = resultCount;
+        count = 0;
+        ble_gap_start_scan(observerProcessScanResult, this, nullptr);
+        return count;
+    }
+
+    int scan(BleScanResult* results, size_t resultCount, uint16_t timeout) {
+        hal_ble_scan_params_t params;
+        ble_gap_get_scan_parameters(&params, nullptr);
+        params.timeout = timeout;
+        ble_gap_set_scan_parameters(&params, nullptr);
+        return scan(results, resultCount);
+    }
+
+    int scan(BleScanResult* results, size_t resultCount, const BleScanParams& params) {
+        ble_gap_set_scan_parameters(&params, nullptr);
+        return scan(results, resultCount);
+    }
+
+    int stopScanning(void) {
+        return ble_gap_stop_scan();
+    }
+
+    static void observerProcessScanResult(const hal_ble_gap_on_scan_result_evt_t* event, void* context) {
+        BleObserverImpl* impl = static_cast<BleObserverImpl*>(context);
+        BleScanResult result;
+        result.address = event->peer_addr;
+        result.rssi = event->rssi;
+        result.scanResponse.set(event->sr_data, event->sr_data_len);
+        result.advertisingData.set(event->adv_data, event->adv_data_len);
+
+        if (impl->callback != nullptr) {
+            impl->callback(&result);
+        } else if (impl->results != nullptr) {
+            if (impl->count < impl->targetCount) {
+                impl->results[impl->count++] = result;
+                if (impl->count >= impl->targetCount) {
+                    impl->stopScanning();
+                }
+            }
+        }
+    }
+};
+
+
+/*******************************************************
+ * BlePeripheralImpl definition
+ */
+class BlePeripheralImpl {
+public:
+    BleConnectionParams ppcp;
```

Comment:
Should these be public?

Reply:
Doesn't need this member any more. Just delete it.